rviz_common/src/rviz_common/visualization_frame.cpp

```diff
@@ -326,10 +326,11 @@ void VisualizationFrame::initialize(
   //                render_panel and VisualizationManager
   render_panel_->getRenderWindow()->initialize();
 
-  auto buffer = std::make_shared<tf2_ros::Buffer>();
   auto clock = std::make_shared<rclcpp::Clock>(RCL_ROS_TIME);
-  // TODO(wjwwood): pass the rviz node so tf2 doesn't create it's own...
-  auto tf_listener = std::make_shared<tf2_ros::TransformListener>(*buffer);
+  auto buffer = std::make_shared<tf2_ros::Buffer>();
+  buffer->setUsingDedicatedThread(true);
+  auto tf_listener = std::make_shared<tf2_ros::TransformListener>(
```

Comment:
As long as there's a dedicated thread making sure to spin so that incoming messages are processed this will be fine.